src/CameraGstreamer.cpp

```diff
@@ -217,6 +218,15 @@ void CameraGstreamer::onVideoFrame( GstVideoFrame *frame ) {
     int currBufferIndex = (_readyToUseBuffer + 1) % RING;
 
     _lastCapturedTimestamp = std::chrono::steady_clock::now();
+
+    const size_t frameSize = GST_VIDEO_FRAME_SIZE(frame);
+    if ( _bufferSize < frameSize ) {
```

Comment:
Why are we allowing frames smaller than our allocated buffer? That sounds like a mistake as well I think.

Reply:
Also, why not make the conditional part of the assert and be done with it?
Is it a logging issue?